server/channels/app/import_functions_test.go

```diff
@@ -2013,579 +2013,749 @@ func TestImportimportMultiplePostLines(t *testing.T) {
+					},
 				},
 			},
-		},
-		LineNumber: 1,
-	}
-
-	errLine, err = th.App.importMultiplePostLines(th.Context, []imports.LineImportWorkerData{data}, false, true)
-	assert.Nil(t, err, "Expected success.")
-	assert.Equal(t, 0, errLine)
+			LineNumber: 1,
+		}
 
-	AssertAllPostsCount(t, th.App, initialPostCount, 5, team.Id)
+		errLine, err2 := th.App.importMultiplePostLines(th.Context, []imports.LineImportWorkerData{data}, false, true)
+		assert.Nil(t, err2, "Expected success.")
+		assert.Equal(t, 0, errLine)
+		assertionCount++
+		AssertAllPostsCount(t, th.App, initialPostCount, assertionCount, team.Id)
 
-	// Check the post values.
-	posts, nErr = th.App.Srv().Store().Post().GetPostsCreatedAt(channel.Id, flagsTime)
-	require.NoError(t, nErr)
+		// Check the post values.
+		posts, nErr := th.App.Srv().Store().Post().GetPostsCreatedAt(channel.Id, flagsTime)
+		require.NoError(t, nErr)
 
-	require.Len(t, posts, 1, "Unexpected number of posts found.")
+		require.Len(t, posts, 1, "Unexpected number of posts found.")
 
-	post = posts[0]
-	postBool = post.Message != *data.Post.Message || post.CreateAt != *data.Post.CreateAt || post.UserId != user.Id
-	require.False(t, postBool, "Post properties not as expected")
+		post := posts[0]
+		postBool := post.Message != *data.Post.Message || post.CreateAt != *data.Post.CreateAt || post.UserId != user.Id
+		require.False(t, postBool, "Post properties not as expected")
 
-	checkPreference(t, th.App, user.Id, model.PreferenceCategoryFlaggedPost, post.Id, "true")
-	checkPreference(t, th.App, user2.Id, model.PreferenceCategoryFlaggedPost, post.Id, "true")
+		checkPreference(t, th.App, user.Id, model.PreferenceCategoryFlaggedPost, post.Id, "true")
+		checkPreference(t, th.App, user2.Id, model.PreferenceCategoryFlaggedPost, post.Id, "true")
+	})
 
-	// Post with reaction.
-	reactionPostTime := hashtagTime + 2
-	reactionTime := hashtagTime + 3
-	data = imports.LineImportWorkerData{
-		LineImportData: imports.LineImportData{
-			Post: &imports.PostImportData{
-				Team:     &teamName,
-				Channel:  &channelName,
-				User:     &username,
-				Message:  model.NewPointer("Message with reaction"),
-				CreateAt: &reactionPostTime,
-				Reactions: &[]imports.ReactionImportData{{
-					User:      &user2.Username,
-					EmojiName: model.NewPointer("+1"),
-					CreateAt:  &reactionTime,
-				}},
+	t.Run("import new post with reactions", func(t *testing.T) {
+		// Post with reaction.
+		reactionPostTime := hashtagTime + 2
+		reactionTime := hashtagTime + 3
+		data := imports.LineImportWorkerData{
+			LineImportData: imports.LineImportData{
+				Post: &imports.PostImportData{
+					Team:     &teamName,
+					Channel:  &channelName,
+					User:     &username,
+					Message:  model.NewPointer("Message with reaction"),
+					CreateAt: &reactionPostTime,
+					Reactions: &[]imports.ReactionImportData{{
+						User:      &user2.Username,
+						EmojiName: model.NewPointer("+1"),
+						CreateAt:  &reactionTime,
+					}},
```

Comment:
Do we have a case where we're importing multiple different reactions and/or multiple of the same kind of reaction? (just wondering if that's something we'd want to test?)

Reply:
That's definitely a valid case, let me add another reaction to that slice.